subsys/logging/log_core.c

```diff
@@ -544,19 +544,20 @@ bool z_impl_log_process(void)
 	if (IS_ENABLED(CONFIG_LOG_MODE_DEFERRED)) {
 		bool dropped_pend = z_log_dropped_pending();
 		bool unordered_pend = z_log_unordered_pending();
+		uint64_t now = k_uptime_get();
 
-		if ((dropped_pend || unordered_pend) &&
-		   (k_uptime_get() - last_failure_report) > CONFIG_LOG_FAILURE_REPORT_PERIOD) {
+		if ((dropped_pend || unordered_pend) && (now > last_failure_report) &&
+		   ((now - last_failure_report) > CONFIG_LOG_FAILURE_REPORT_PERIOD)) {
 			if (dropped_pend) {
 				dropped_notify();
 			}
 
 			if (unordered_pend) {
 				unordered_notify();
 			}
+			last_failure_report += CONFIG_LOG_FAILURE_REPORT_PERIOD;
 		}
 
-		last_failure_report += CONFIG_LOG_FAILURE_REPORT_PERIOD;
```

Comment:
@nordic-krch This statement can make the `last failure_report` is larger than uptime(`now`).